### PR TITLE
feat: Add react navigation

### LINKED ISSUE #5422: Implement router solution for mobile app
There are only two options: `react-navigation` and `react-native-navigation`. Comparsion is here https://blog.logrocket.com/react-navigation-vs-react-native-navigation/

### PR DESCRIPTION
**Pull request by bouchja1**
Add structure for react navigation:
- root bottom tab navigation
- tab _settings_ includes stack navigation
- types (screens, navigation, routes)
- tab bar stylying based on [figma](https://www.figma.com/file/Z6AGVUmKQzLNtDozFamW7f/s2-Mobile)
- iOS simulator warning _Module TrezorTransportNative requires main queue setup since it overrides init but does not implement requiresMainQueueSetup_ [fix](https://github.com/trezor/trezor-suite/pull/5455/files#diff-e6a2ba7d7340399c6d3727b466e8f6b06ba8a4752d7340e9c0911f8e0397d51d)

@Nodonisko Please let's discuss about the solution and possible improvements/changes.

[video](https://user-images.githubusercontent.com/2227592/169706671-1f5f068d-9a67-466f-b18c-57d4193b2c89.mov)

Closes https://github.com/trezor/trezor-suite/issues/5422

### CODE REVIEW

File: packages/suite-native/src/navigation/utils/navigation.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { createNavigationContainerRef } from '@react-navigation/native';
```
Comment (Nodonisko) [inline packages/suite-native/src/navigation/utils/navigation.ts]: keep it flat, not reason to have this in extra folder

Comment (Nodonisko) [reply packages/suite-native/src/navigation/utils/navigation.ts]: I not sure if this file even make sense, from what I see it makes much more sense to move these things do place where they are used aka TabBar and Navigation because they are clearly coupled with these files anyway

File: packages/transport-native/ios/TrezorTransportNative.m [hunk 2]
```diff
@@ -6,4 +6,10 @@ @interface RCT_EXTERN_MODULE(TrezorTransportNative, NSObject)
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)
 
+// https://github.com/facebook/react-native/issues/23755#issuecomment-474235632
```
Comment (Nodonisko) [inline packages/transport-native/ios/TrezorTransportNative.m:9]: why this? the link refers to some ImagePicker issue which seem irrelevant for us

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: packages/suite-native-navigation/src/TabBar.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+export const bottomTabOptions: Record<string, { icon: IconType }> = {
+    [RouteTabs.Home]: { icon: 'home' },
+    [RouteTabs.Accounts]: { icon: 'accounts' },
+    [RouteTabs.Action]: { icon: 'action' },
+    [RouteTabs.Prices]: { icon: 'prices' },
+    [RouteTabs.SettingsStack]: { icon: 'settings' },
+};
+
+export const TabBar = ({ state, descriptors, navigation }: BottomTabBarProps) => {
```
Comment (juriczech) [inline packages/suite-native-navigation/src/TabBar.tsx]: Just curious: why not use the `useNavigation` hook?

File: packages/suite-native-navigation/src/TabBar.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                            handleTabPress();
+                        }}
+                        key={route.key}
+                    >
+                        <Icon
+                            type={bottomTabOptions[route.name].icon}
+                            size="big"
+                            color={isFocused ? 'black' : 'gray500'}
+                        />
+                    </TouchableOpacity>
```
Comment (juriczech) [inline packages/suite-native-navigation/src/TabBar.tsx]: This looks like it could be a standalone component maybe?

File: packages/suite-native/src/navigation/utils/navigation.ts (hunk 1 above)
Comment (bouchja1) [reply packages/suite-native/src/navigation/utils/navigation.ts]: This was completely refactored (packages, navigation creation coupled with components etc.) and we are starting again from https://github.com/trezor/trezor-suite/pull/5455/commits/d73f4bb497b134b55b30e5d543dfe8be1f1f8c87

File: packages/suite-native-navigation/src/TabBar.tsx (hunk 3 above)
Comment (bouchja1) [reply packages/suite-native-navigation/src/TabBar.tsx]: I had the same question at first, but according to documentation:

> Note that you cannot use the useNavigation hook inside the tabBar since useNavigation is only available inside screens. You get a navigation prop for your tabBar which you can use instead.

File: packages/suite-native-navigation/src/TabBar.tsx (hunk 4 above)
Comment (bouchja1) [reply packages/suite-native-navigation/src/TabBar.tsx]: Do you mean to create something like TabBarIcon and move it to the atoms package? Is there a true reason for this? I guess it won't be reusable at all anywhere except this one and only tab bar.

File: packages/suite-native-navigation/src/TabBar.tsx (hunk 3 above)
Comment (juriczech) [reply packages/suite-native-navigation/src/TabBar.tsx]: Magic JS 😅

Thanks for explaining

File: packages/suite-native-navigation/src/types.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import type { BottomTabScreenProps, BottomTabNavigationProp } from '@react-navigation/bottom-tabs';
+import type { StackScreenProps, StackNavigationProp } from '@react-navigation/stack';
+import { NavigatorScreenParams, ParamListBase } from '@react-navigation/native';
+import { SettingsStackParamList } from '@suite-native/settings';
+import { RouteTabs } from './routes';
+import type { RouteProp } from '@react-navigation/native';
+
+export type RootTabParamList = {
```
Comment (Nodonisko) [inline packages/suite-native-navigation/src/types.tsx]: please move this to routes file, these are types but directly related to routes so it will be better to keep them together

File: packages/suite-native-settings/src/navigation/types.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { SettingsRouteStack } from './routes';
+
+// To type check our route name and params - mappings for route name to the params of the route
+export type SettingsStackParamList = {
```
Comment (Nodonisko) [inline packages/suite-native-settings/src/navigation/types.ts]: please move this to routes

File: packages/suite-native-navigation/src/TabBar.tsx (hunk 4 above)
Comment (juriczech) [reply packages/suite-native-navigation/src/TabBar.tsx]: To me it seems like this could be a TapBarList (this cmp) + TapBar cmp which would be something like 
```
state.routes.map((route, index) => {
    // some code, destructuring etc
    return <TabBar {...props} />
})
```

It's not really going to be that much about reusability but imho the render method would be much more cleaner. In this component everything about the general tabs will be very clear and then you have a specific component of <TabBar />. My OCD likes this approach much more to have mapped items as a standalone component but that's just my view. Maybe you prefer to have it all inside one component like this and I just need to supress my OCD feelings 😀

File: packages/suite-native-navigation/src/screens/DummyScreen.tsx [hunk 7]
```diff
@@ -12,7 +12,7 @@ const backgroundStyle = prepareNativeStyle<{ isDarkMode: boolean }>(
     }),
 );
 
-export const Home = () => {
+export const DummyScreen = () => {
```
Comment (Nodonisko) [inline packages/suite-native-navigation/src/screens/DummyScreen.tsx]: Please rename this to `DemoScreen` and rename this whole folder to `dummyScreens` to make it obvious that no one should add anything serious here

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: packages/suite-native-navigation/src/TabBar.tsx (hunk 4 above)
Comment (bouchja1) [reply packages/suite-native-navigation/src/TabBar.tsx]: Ok, I understand your thoughts. I cleaned it up a little in https://github.com/trezor/trezor-suite/pull/5455/commits/586add7a095903bc6b92f49f63f61c3c7927d312

File: packages/suite-native-navigation/src/types.tsx (hunk 5 above)
Comment (bouchja1) [reply packages/suite-native-navigation/src/types.tsx]: moved

File: packages/suite-native-navigation/src/screens/DummyScreen.tsx (hunk 7 above)
Comment (bouchja1) [reply packages/suite-native-navigation/src/screens/DummyScreen.tsx]: demo, dummy, hodinky, holinky... :D Ok, renamed in https://github.com/trezor/trezor-suite/commit/586add7a095903bc6b92f49f63f61c3c7927d312

**Review by juriczech — APPROVED** [review]

File: packages/suite-native-navigation-root/package.json [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    "dependencies": {
+        "@react-navigation/bottom-tabs": "^6.3.1",
+        "@react-navigation/native": "^6.0.10",
+        "@suite-native/settings": "*",
+        "@suite-native/navigation": "*",
+        "@trezor/atoms": "*",
+        "@trezor/styles": "*",
+        "react": "^17.0.2",
+        "react-native": "0.68.2",
+        "react-native-gesture-handler": "^2.4.2",
```
Comment (Nodonisko) [inline packages/suite-native-navigation-root/package.json]: is this library used in this package? I believe it's used only in suite-native

File: packages/suite-native-navigation-root/package.json [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        "react-native-screens": "^3.13.1",
+        "react-native-safe-area-context": "^4.2.5"
```
Comment (Nodonisko) [inline packages/suite-native-navigation-root/package.json]: is this library used in this lib?

File: packages/suite-native-navigation/package.json [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+        "type-check": "tsc --build"
+    },
+    "dependencies": {
+        "@react-navigation/bottom-tabs": "^6.3.1",
+        "@react-navigation/native": "^6.0.10",
+        "@trezor/atoms": "*",
+        "@trezor/styles": "*",
+        "react": "^17.0.2",
+        "react-native": "0.68.2",
+        "react-native-gesture-handler": "^2.4.2",
```
Comment (Nodonisko) [inline packages/suite-native-navigation/package.json]: is it used?

File: packages/suite-native-navigation/package.json [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+        "react-native-screens": "^3.13.1",
+        "react-native-safe-area-context": "^4.2.5"
```
Comment (Nodonisko) [inline packages/suite-native-navigation/package.json]: is it used?

File: packages/suite-native-navigation/src/TabBarItem.tsx [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+    onPress: () => void;
+    icon: IconType;
+};
+
+export const TabBarItem = ({ isFocused, onPress, icon }: TabBarItemProps) => (
+    <TouchableOpacity
+        accessibilityRole="button"
+        accessibilityState={isFocused ? { selected: true } : {}}
+        onPress={() => {
+            // Calling in a function prevents Animation warning in iOS simulator (see https://github.com/react-navigation/react-navigation/issues/7839#issuecomment-829438793 for more details)
```
Comment (Nodonisko) [inline packages/suite-native-navigation/src/TabBarItem.tsx]: nit: can you split comment to two lines? :D

File: packages/suite-native-settings/package.json [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    "dependencies": {
+        "@react-navigation/native": "^6.0.10",
+        "@react-navigation/stack": "^6.2.1",
+        "@suite-native/navigation": "*",
+        "@trezor/atoms": "*",
+        "@trezor/styles": "*",
+        "react": "^17.0.2",
+        "react-native": "0.68.2",
+        "react-native-gesture-handler": "^2.4.2",
```
Comment (Nodonisko) [inline packages/suite-native-settings/package.json]: is it used?

File: packages/suite-native-settings/package.json [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+        "react-native-screens": "^3.13.1",
+        "react-native-safe-area-context": "^4.2.5"
```
Comment (Nodonisko) [inline packages/suite-native-settings/package.json]: is it used?

File: packages/suite-native-settings/src/navigation/routes.ts [hunk 15]
```diff
@@ -0,0 +1,9 @@
+export enum SettingsRouteStack {
+    Settings = 'Settings',
+    SettingsDetail = 'SettingsDetail',
+}
+
+export type SettingsStackParamList = {
+    [SettingsRouteStack.Settings]: undefined;
+    [SettingsRouteStack.SettingsDetail]: { msg: string };
```
Comment (Nodonisko) [inline packages/suite-native-settings/src/navigation/routes.ts]: Please don't use shortcuts at all msg => message, I know this is just example but then someone will copy that.

File: packages/suite-native/package.json [hunk 16]
```diff
@@ -14,15 +14,21 @@
         "pods": "npx pod-install"
     },
     "dependencies": {
+        "@react-native-masked-view/masked-view": "^0.2.6",
```
Comment (Nodonisko) [inline packages/suite-native/package.json]: is this used anywhere?

File: packages/suite-native-settings/src/navigation/routes.ts [hunk 17]
```diff
@@ -0,0 +1,9 @@
+export enum SettingsRouteStack {
```
Comment (Nodonisko) [inline packages/suite-native-settings/src/navigation/routes.ts]: nit: SettingsStackRoutes

File: packages/suite-native-navigation-root/src/navigation/routes.ts [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+
+export enum RouteTabs {
+    Home = 'Home',
+    Accounts = 'Accounts',
+    Action = 'Action',
+    Prices = 'Prices',
+    SettingsStack = 'SettingsStack',
+}
+
+export type RootTabParamList = {
```
Comment (Nodonisko) [inline packages/suite-native-navigation-root/src/navigation/routes.ts]: Tab => Tab**s**

File: packages/suite-native-navigation-root/src/navigation/routes.ts [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+    [RouteTabs.Home]: undefined;
+    [RouteTabs.Accounts]: undefined;
+    [RouteTabs.Action]: undefined;
+    [RouteTabs.Prices]: undefined;
+    [RouteTabs.SettingsStack]: NavigatorScreenParams<SettingsStackParamList>;
+};
+
+export const rootTabOptions: TabOption = {
```
Comment (Nodonisko) [inline packages/suite-native-navigation-root/src/navigation/routes.ts]: Tab => Tab**s**

**Review by Nodonisko — CHANGES_REQUESTED** [review]

**Review by Nodonisko — APPROVED** [review]
